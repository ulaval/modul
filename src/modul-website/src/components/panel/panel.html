<transition name="mw--is">
    <section class="mw-panel"
             v-if="open"
             tabindex="0">
        <m-icon-button class="mw-panel__close-button"
                       @click="closeMenu"
                       icon-name="m-svg__close-clear"
                       skin="light"
                       icon-size="20px">
        </m-icon-button>
        <slot></slot>
    </section>
</transition>
