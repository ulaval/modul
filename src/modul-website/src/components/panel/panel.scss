@import '../../styles/commons';

$modul-menu--list-max-width: 300px;
$modul-menu-offset-left: $m-padding--s + 2;
$modul-menu-beta-width: 24px;
$modul-menu-transition-duration: 0.7s;

.mw-panel {
    position: fixed;
    top: $mw-header-height--s;
    right: 0;
    bottom: 0;
    left: 0;
    z-index: 2;
    overflow-y: auto;
    overflow-x: hidden;
    background: $modul-color--black-grey;
    color: $modul-color--grey-light;
    outline: none;
    box-shadow: $m-box-shadow--l;

    @media (min-width: #{$m-mq--min--s}) {
        top: $mw-header-height;
    }

    @media (min-width: #{$m-mq--min--s}) {
        padding: $m-padding--xl $m-padding 0 $m-padding;
    }

    @media (min-width: #{$m-mq--min--l}) {
        padding-left: $m-padding--xl*2;
        padding-right: $m-padding--xl*2;
    }

    &.mw--is {
        &-enter-active {
            transition: transform #{$modul-CSS_ANIMATION_MENU_DURATION} ease;
        }

        &-leave-active {
            transition: transform #{$modul-CSS_ANIMATION_MENU_DURATION} ease;
        }

        &-leave-active,
        &-enter {
            transform: translate3d(0, -102%, 0);
        }
    }

    &__close-button {
        position: absolute;
        top: 0;
        right: 0;

        @media (max-width: #{$m-mq--min--s}) {
            display: none;
        }
    }
}
