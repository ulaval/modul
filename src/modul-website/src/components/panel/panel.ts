import { MediaQueries } from '@ulaval/modul-components/dist/mixins/media-queries/media-queries';
import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Emit, Prop } from 'vue-property-decorator';
import { ModulWebsite } from '../modul-website';
import WithRender from './panel.html?style=./panel.scss';

@WithRender
@Component({
    mixins: [MediaQueries]
})
export class MWPanel extends ModulWebsite {
    @Prop({ default: false })
    public open: boolean;

    @Emit('close')
    closeMenu() { }
}

export const MWPANEL_NAME: string = 'mw-panel';

const MWPanelPlugin: PluginObject<any> = {
    install(v, options) {
        v.component(MWPANEL_NAME, MWPanel);
    }
};

export default MWPanelPlugin;
