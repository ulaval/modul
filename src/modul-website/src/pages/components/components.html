<div class="mw-components">
    <m-expandable-layout class="mw-components__body"
                         :open="showMenu"
                         panel-width="320px">
        <div class="mw-components__page">
            <m-icon-button class="mw-components__menu-button"
                           @click="toggleMenu()"
                           :icon-name="showMenu ? 'm-svg__close-clear' : 'm-standard-edit'"
                           skin="light"
                           icon-size="20px">
            </m-icon-button>

            <div class="mw-components__content">
                <mw-component-details :component="component"></mw-component-details>
            </div>
        </div>
        <template slot="panel">
            <div class="mw-components__sidemenu">

                <div class="mw-components__search">
                    <m-searchfield></m-searchfield>
                </div>
                <div class="mw-components__sidemenu-body">
                    <div class="mw-components__sidemenu-title">
                        <m-i18n k="website:components-ui"></m-i18n>
                    </div>
                    <m-accordion skin="plain"
                                 icon-position="right"
                                 icon-size="small"
                                 v-for="categoryId in componentCategoryIds"
                                 :key="categoryId"
                                 :open="urlMatchPath(categoryId)">
                        <div class="mw-components__sidemenu-sub-title"
                             slot="header">
                            <m-i18n :k="'categories:'+categoryId"></m-i18n>
                        </div>
                        <div>
                            <div class="mw-components__sidemenu-item"
                                 v-for="component in getCategoryComponents(categoryId)"
                                 :key="component.url">
                                <router-link :to="getRouterIndex(component.url)"
                                             @click.native="onSideMenuSelection()"><span>
                                        {{component.name}}
                                    </span>
                                    <span v-if="component.beta">
                                        <span class="modul-logo__beta mw-components__beta">
                                            <m-i18n k="website:beta"></m-i18n>
                                        </span>
                                    </span>
                                </router-link>
                            </div>
                        </div>
                    </m-accordion>
                </div>
            </div>
        </template>
    </m-expandable-layout>
</div>
