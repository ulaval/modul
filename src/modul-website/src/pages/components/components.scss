@import '../../styles/commons';

.mw-component-details {
    width: 100%;

    &__component-navigation {
        display: flex;
        margin-top: 54px;
        color: $m-color--interactive-light;
    }

    &__component-navigation-next,
    &__component-navigation-prev {
        line-height: 24px;
        position: relative;
        padding: 1rem;
        flex-grow: 1;
        border: 1px solid;
        text-align: center;
        text-decoration: none;
        font-weight: $m-font-weight--semi-bold;
    }

    &__component-navigation-next--icon {
        position: absolute;
        right: 16px;
    }

    &__component-navigation-prev--icon {
        position: absolute;
        left: 16px;
    }
}
