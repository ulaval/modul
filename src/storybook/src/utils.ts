export const modulComponentsHierarchyRootSeparator: string = 'modul-components|';
export const modulWebsiteHierarchyRootSeparator: string = 'modul-website|';
