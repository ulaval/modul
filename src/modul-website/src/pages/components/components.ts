
import { MWExpandablePanel } from '@/components/expendable-panel/expandable-panel';
import { ModulWebsite } from '@/components/modul-website';
import { ComponentMeta } from '@/content/components.meta.loader';
import _ from 'lodash';
import Vue from 'vue';
import Component from 'vue-class-component';
import { Prop, Watch } from 'vue-property-decorator';
import { MWComponent } from './component/component-details';
import WithRender from './components.html?style=./components.scss';

@WithRender
@Component({
    components: { 'mw-component-details': MWComponent }
})
export class MWComponentsPage extends ModulWebsite {

    @Prop()
    component: ComponentMeta;

    $refs: {
        expandablePanel: MWExpandablePanel;
        prevLink: Vue;
        nextLink: Vue;
    };

    fullPath: string;

    get componentCategoryIds(): string[] {
        return _.sortBy(this.$meta.categories, name => _.deburr(this.$i18n.translate(`categories:${name}`)));
    }

    get componentsNavigationOrder(): ComponentMeta[] {
        let components: ComponentMeta[] = [];
        this.componentCategoryIds.forEach(category => {
            this.$meta.componentState[category].forEach((componentMeta: ComponentMeta) => {
                components.push(componentMeta);
            });
        });
        return components;
    }

    get currentComponentNavigationIndex(): number {
        return _.findIndex(this.componentsNavigationOrder, (c: ComponentMeta) => {
            return this.component.name === c.name;
        });
    }

    get prevComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the first item return the last
        if (index - 1 < 0) {
            return this.componentsNavigationOrder[this.componentsNavigationOrder.length - 1];
        } else {
            return this.componentsNavigationOrder[index - 1];
        }
    }

    get nextComponent(): ComponentMeta {
        let index = this.currentComponentNavigationIndex;

        // if this is the last item return the first
        if (index + 1 >= this.componentsNavigationOrder.length - 1) {
            return this.componentsNavigationOrder[0];
        } else {
            return this.componentsNavigationOrder[index + 1];
        }
    }

    getRouterIndex(tag: string): string {
        return this.$routerIndex.for(tag);
    }

    getCategoryComponents(category): ComponentMeta[] {
        return this.$meta.componentState[category];
    }

    urlMatchPath(categoryId: string) {
        let path: string = this.$i18n.translate(`categories:${categoryId}-route`);
        return this.fullPath.includes(`/${path}`);
    }

    onSideMenuSelection() {
        this.$refs.expandablePanel.onSideMenuSelection();
    }

    onPrevLinkClick() {
        (this.$refs.prevLink.$el as HTMLElement).blur();
    }

    onNextLinkClick() {
        (this.$refs.nextLink.$el as HTMLElement).blur();
    }

    @Watch('$route', { immediate: true })
    private routeChanged(): void {
        this.fullPath = this.$route.fullPath;
    }
}
