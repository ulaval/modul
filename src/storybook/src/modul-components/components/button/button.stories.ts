import { actions } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs';
import { BUTTON_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';


export default {
    title: `${modulComponentsHierarchyRootSeparator}${BUTTON_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    props: {
        text: {
            default: text('Text', 'A Button')
        }
    },
    methods: actions(
        'click',
        'focus',
        'blur'
    ),
    template: '<m-button @click="click" @focus="focus" @blur="blur">{{ text }}</m-button>'
});

defaultStory.story = {
    name: 'default'
};

export const disabled = () => '<m-button :disabled="true">A Button</m-button>';

export const waiting = () => '<m-button :waiting="true">A Button</m-button>';

export const fullsize = () => '<m-button fullSize="true">A Button</m-button>';

export const precision = () => '<m-button >A Button <template slot="precision">Button precision</template></m-button>';

export const submit = () => '<m-button type="submit">A Button</m-button>';

export const reset = () => '<m-button type="reset">A Button</m-button>';

export const secondaryDefault = () => '<m-button skin="secondary">A Button</m-button>';

export const secondaryDisabled = () => '<m-button skin="secondary" :disabled="true">A Button</m-button>';

export const secondaryWaiting = () => '<m-button skin="secondary" :waiting="true">A Button</m-button>';

export const secondaryFullSize = () => '<m-button :fullSize="true" skin="secondary">A Button</m-button>';

export const secondaryIcon = () => '<m-button icon-name="m-svg__close-clear" skin="secondary">A Button</m-button>';

export const secondaryIcon20Px = () => '<m-button icon-name="m-svg__close-clear" icon-size="20px" skin="secondary">A Button</m-button>';

export const iconPositionRight = () => '<m-button icon-name="m-svg__close-clear" icon-position="right" skin="secondary">A Button</m-button>';

export const secondaryPrecision = () => '<m-button skin="secondary" >A Button <template slot="precision">Button precision</template></m-button>';

export const icon = () => '<m-button icon-name="m-svg__close-clear">A Button</m-button>';

export const iconDisabled = () => '<m-button :disabled="true" icon-name="m-svg__close-clear">A Button</m-button>';

export const iconWaiting = () => '<m-button icon-name="m-svg__close-clear" waiting="true">A Button</m-button>';

export const iconFullSize = () => '<m-button :fullSize="true" icon-name="m-svg__close-clear">A Button</m-button>';

export const icon20Px = () => '<m-button icon-name="m-svg__close-clear" icon-size="20px">A Button</m-button>';

export const iconPrecision = () => '<m-button icon-name="m-svg__close-clear">A Button <template slot="precision">' +
    'Button precision</template></m-button>';
