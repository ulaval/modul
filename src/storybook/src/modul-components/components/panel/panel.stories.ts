import { storiesOf } from '@storybook/vue';
import { MButtonSkin } from '@ulaval/modul-components/dist/components/button/button';
import { PANEL_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MPanelSkin } from '@ulaval/modul-components/dist/components/panel/panel';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';


storiesOf(`${modulComponentsHierarchyRootSeparator}${PANEL_NAME}`, module)
    .add('default', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel>
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add('highlighted="true"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :highlighted="true">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add('shadow="true"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :shadow="true">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }));

storiesOf(`${modulComponentsHierarchyRootSeparator}${PANEL_NAME}/skin`, module)
    .add(`skin="${MPanelSkin.Light}"`, () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel skin="${MPanelSkin.Light}">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add(`skin="${MPanelSkin.Dark}"`, () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel skin="${MPanelSkin.Dark}">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add(`skin="${MPanelSkin.Darker}"`, () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel skin="${MPanelSkin.Darker}">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }));

storiesOf(`${modulComponentsHierarchyRootSeparator}${PANEL_NAME}/border`, module)
    .add('border="false"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :border="false">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add('border-large="true"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :border-large="true">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }));

storiesOf(`${modulComponentsHierarchyRootSeparator}${PANEL_NAME}/padding`, module)
    .add('padding="false"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :padding="false">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add('padding-large="true"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :padding-large="true">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add('padding-header="false"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :padding-header="false">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add('padding-body="false"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :padding-body="false">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }))
    .add('padding-footer="false"', () => ({
        template: `<div class="m-u--padding--m">
                    <m-panel :padding-footer="false">
                        <h2 slot="header">I am a header</h2>
                        <p>Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p slot="footer">I am footer</p>
                    </m-panel>
                </div>`
    }));

storiesOf(`${modulComponentsHierarchyRootSeparator}${PANEL_NAME}/$slots example`, module)
    .add('$slots reactivity', () => ({
        data: () => ({
            hasHeader: true,
            hasBody: true,
            hasFooter: true,
            getSkinButton(value: boolean): MButtonSkin {
                return value ? MButtonSkin.Primary : MButtonSkin.Secondary;
            }
        }),
        template: `<div class="m-u--padding--m">
                    <m-panel>
                        <h2 v-if="hasHeader" slot="header">I am a header</h2>
                        <p v-if="hasBody">Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.</p>
                        <p v-if="hasFooter" slot="footer">I am footer</p>
                    </m-panel>
                    <m-button class="m-u--margin-top" :skin="getSkinButton(hasHeader)"@click="hasHeader = !hasHeader">Toogle $slot.header</m-button>
                    <m-button class="m-u--margin-top":skin="getSkinButton(hasBody)"@click="hasBody = !hasBody">Toogle $slot.default</m-button>
                    <m-button class="m-u--margin-top":skin="getSkinButton(hasFooter)"@click="hasFooter = !hasFooter">Toogle $slot.footer</m-button>
                </div>`
    }));
