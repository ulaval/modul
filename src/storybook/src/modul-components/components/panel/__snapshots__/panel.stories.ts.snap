// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-panel default 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel highlighted="true" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border m--is-highlighted"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel shadow="true" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-shadow m--has-border"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/$slots example $slots reactivity 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
   
  <button
    class="m-button m-u--margin-top m--is-skin-primary"
    type="button"
  >
    <!---->
     
    <!---->
     
    <span
      class="m-button__text"
    >
      Toogle $slot.header 
      <!---->
    </span>
     
    <!---->
     
    <!---->
  </button>
   
  <button
    class="m-button m-u--margin-top m--is-skin-primary"
    type="button"
  >
    <!---->
     
    <!---->
     
    <span
      class="m-button__text"
    >
      Toogle $slot.default 
      <!---->
    </span>
     
    <!---->
     
    <!---->
  </button>
   
  <button
    class="m-button m-u--margin-top m--is-skin-primary"
    type="button"
  >
    <!---->
     
    <!---->
     
    <span
      class="m-button__text"
    >
      Toogle $slot.footer 
      <!---->
    </span>
     
    <!---->
     
    <!---->
  </button>
</div>
`;

exports[`Storyshots modul-components|m-panel/border border="false" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/border border-large="true" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border m--has-border-large"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/padding padding="false" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border"
  >
    <header
      class="m-panel__header m--no-padding"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--no-padding"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer m--no-padding"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/padding padding-body="false" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--no-padding"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/padding padding-footer="false" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer m--no-padding"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/padding padding-header="false" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border"
  >
    <header
      class="m-panel__header m--no-padding"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/padding padding-large="true" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border m--has-padding-large"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/skin skin="dark" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-dark m--has-border"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/skin skin="darker" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-darker m--has-border"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;

exports[`Storyshots modul-components|m-panel/skin skin="light" 1`] = `
<div
  class="m-u--padding--m"
>
  <article
    class="m-panel m--is-skin-light m--has-border"
  >
    <header
      class="m-panel__header"
    >
      <div
        class="m-panel__header__content"
      >
        <h2>
          I am a header
        </h2>
      </div>
       
      <!---->
       
      <!---->
    </header>
     
    <div
      class="m-panel__body m--has-padding-bottom"
    >
       
      <p>
        Lorem ipsum dolor, sit amet consectetur adipisicing elit.Earum voluptas quis corporis fugit nobis beatae vel, quia at consequuntur.
      </p>
       
    </div>
     
    <footer
      class="m-panel__footer"
    >
      <p>
        I am footer
      </p>
    </footer>
  </article>
</div>
`;
